scripts/Diffraction/isis_powder/pearl.py

```diff
@@ -96,6 +96,20 @@ def _generate_out_file_paths(self, run_details):
             output_file_paths["output_name"] = output_file_paths["output_name"] + file_ext.replace(".", "_")
         return output_file_paths
 
+    def get_trans_module_indices(self):
+        default_imods = list(range(9))  # all modules 1-9 in transverse banks (tth~90 deg)
+        default_mod_nums_str = ""
+        if self._inst_settings.trans_mod_nums and self._inst_settings.focus_mode == "trans_custom":
+            mod_nums = common.generate_run_numbers(run_number_string=self._inst_settings.trans_mod_nums)
+            imods = [int(mod_num - 1) for mod_num in set(mod_nums) if 0 < mod_num < 10]  # remove invalid/duplicates
+            if len(imods) < len(mod_nums):
+                # catches case where no indices in correct range as len(mod_nums) > 1 in this branch
+                logger.warning("Invalid or duplicate modules in trans_mod_nums - using all modules 1-9")
+                return default_imods, default_mod_nums_str
+            return imods, self._inst_settings.trans_mod_nums
+        else:
+            return default_imods, default_mod_nums_str
+
```

Comment:
Using the "return early" pattern in this function might help with readability, although I know this is subjective so feel free to ignore this comment! https://medium.com/swlh/return-early-pattern-3d18a41bba8